packages/alsatian/core/running/callback-listener-informer.ts

```diff
@@ -0,0 +1,9 @@
+import {CallbackFunction} from "../events/generic-callback";
+
+export class ListenerInformer {
+	public async informListeners<B, T extends CallbackFunction<B>>(callbackFunctions: Array<T>, parameter: B) {
```

Comment:
There is a built in `Parameters` type for TypeScript that may be useful for this use case and means we can probably drop the generics from `CallbackFunction` to make it simpler

Reply:
Ok never used this :) I will do some research about this sounds cool.

Reply:
Ok i analysed this and its not clear to me how this could help here. The generics are mainly there to make things type safe so that you can only call the informListeners method with a valid set of list of callbacks and correct parameter for the list of callbacks.

Reply:
What this means I believe is that we can simplify the code as below

```typescript
type CallbackFunction = (...args: Array<any>) => void;

public async informListeners<F extends CallbackFunction>(callbackFunctions: Array<F>, ...parameters: Parameters<F>) {
```

Now if we want to enforce a single argument we can write the following

```typescript
type CallbackFunction = (arg: any) => void;

public async informListeners<F extends CallbackFunction>(callbackFunctions: Array<F>, ...parameters: Parameters<F>) {

// OR this to avoid using spread but personally spread feels a bit nicer
public async informListeners<F extends CallbackFunction>(callbackFunctions: Array<F>, parameter: Parameters<F>[0]) {
```